src/internal/item-container/index.tsx

```diff
@@ -43,20 +46,84 @@ export interface ItemContainerRef {
 }
 
 interface ItemContextType {
+  /**
+   * Flag indicating if a drag or resize interaction is currently active.
+   */
   isActive: boolean;
+  /**
+   * Flag indicating if the item is currently hidden.
+   * (When a board item is moved from the palette to the board and the transition is not submitted)
+   */
+  isHidden: boolean;
   dragHandle: {
-    ref: RefObject<HTMLButtonElement>;
+    /**
+     * Ref to the drag button. Used to focus the drag handle when moving an item
+     * from the palette to the board via keyboard or UAP actions.
+     */
+    ref: RefObject<HTMLDivElement>;
+    /**
+     * Listen to pointerDown events on the drag handle.
+     * Used to start a transition and attach global event handlers.
+     */
     onPointerDown(event: ReactPointerEvent): void;
+    /**
+     * Listen to keyDown events on the drag handle.
+     */
     onKeyDown(event: KeyboardEvent): void;
-    isActive: boolean;
+    /**
+     * Flag indicating if a pointer interaction is active.
+     */
+    isActivePointer: boolean;
+    /**
+     * Flag indicating if UAP interaction mode is active.
+     */
+    isActiveUap: boolean;
```

Comment:
Is there a separate active state for keyboard, too?

Would it make sense to replace the boolean flags with a single state like `activeState: null | 'pointer' | 'keyboard' | 'uap'`? The benefit is that we won't need to worry about the possibility of multiple boolean flags to be set at the same time.

Reply:
No, it's just for `pointer` and `uap`. I like your suggestion about simplification. Updated it!